pkg/cloud/rgraph/rnode/backendservice/type_trait.go

```diff
@@ -43,11 +43,136 @@ func (*typeTrait) FieldTraits(meta.Version) *api.FieldTraits {
 	dt.OutputOnly(api.Path{}.Pointer().Field("SecurityPolicy"))
 	dt.OutputOnly(api.Path{}.Pointer().Field("SelfLink"))
 
-	dt.OutputOnly(api.Path{}.Pointer().Field("Iap").Field("Oauth2ClientSecretSha256"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Iap"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Iap").Pointer().Field("Enabled"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Iap").Pointer().Field("Oauth2ClientId"))
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Iap").Pointer().Field("Oauth2ClientSecret"))
+	dt.OutputOnly(api.Path{}.Pointer().Field("Iap").Pointer().Field("Oauth2ClientSecretSha256"))
+
+	dt.AllowZeroValue(api.Path{}.Pointer().Field("Port"))
```

Comment:
Shouldn't it be OutputOnly if it is deprecated?

Reply:
This is a good question :) For me deprecated doesn't mean that you cannot use this but just shouldn't.

Reply:
I think deprecated or not should not affect the type trait -- deprecation is says "it still works, but please don't use it" -- but if we need to continue to use it, then that's what it is...